server/raft_transport.go

```diff
@@ -217,6 +217,10 @@ func (n *natsStreamLayer) newNATSConn(address string) *natsConn {
 // performing a handshake over NATS which establishes unique inboxes at each
 // endpoint for streaming data.
 func (n *natsStreamLayer) Dial(address raft.ServerAddress, timeout time.Duration) (net.Conn, error) {
+	if !n.conn.IsConnected() {
+		return nil, errors.New("stan: raft dial failed since not connected to NATS")
+	}
+
```

Comment:
Some notes around here, `n.conn.Flush()` will allow the reconnection to take up to 60 seconds, but within that time `Dial` will be called currently it seems around every second during the reconnection, so once the reconnection happens all the accumulated `Dial` calls would be unblocked and then the request happens below while it is connected again.

Reply:
I think that we should add to the `natsStreamLayer` a field that will hold the timeout specified when creating the transport. We should then use that timeout for all Flush() calls (replace them with FlushTimeout). We also need to check the return value of the flush and unsub and return error if not nil.
Raft does "pool" the connections when possible, so it is possible that you see some of such inboxes (which correspond to raft connections if you will).

Reply:
I agree with the fixes too, changing to `FlushTimeout` with a shorter timeout sounds better

Reply:
The one that we don't need the actual transport timeout and check for error is in `Close()` which is when RAFT releases a connection and won't reuse it. In this case I think it could be best effort and a constant of say 500ms or 1sec is enough.

Reply:
Do you want to have a go at the changes or do you want me to do it?

Reply:
Looking at another issue today so feel free update, thanks Ivan!

Reply:
I won't probably have a chance to do it today, but will update later on this week. Thanks!